psi4/src/psi4/lib3index/dfhelper.cc

```diff
@@ -316,7 +316,10 @@ void DFHelper::prepare_sparsity() {
 #ifdef _OPENMP
         rank = omp_get_thread_num();
 #endif
-        if (rank) eri[rank] = std::shared_ptr<TwoBodyAOInt>(eri.front()->clone());
+        if (rank) {
+            eri[rank] = std::shared_ptr<TwoBodyAOInt>(eri.front()->clone());
+            if (!eri[rank]->initialized()) eri[rank]->initialize_sieve();
```

Comment:
Does `clone` clone the sieve as well? Should one instead first just make sure `eri.front()` has been `initialized()`?

Reply:
It should, but I will double check. If it does, your suggestion would be a better implementation than what I did here.

Reply:
Yeah, it does copy the sieve as well. So I have changed all initialization instances where a clone is involved, to what you suggested!